analysis_driver/pipelines/demultiplexing.py

```diff
@@ -1,185 +1,175 @@
+            job_name='fastq_filterer',
+            working_dir=self.job_dir,
+            cpus=18,
+            mem=10
+        ).join()
+
+
+class WellDuplicates(DemultiplexingStage):
+    previous_stages = [Setup]
+
+    def _run(self):
+        well_dup_exec = lane_duplicates.WellDuplicates(self.dataset, self.job_dir, self.fastq_dir, self.input_dir)
+        well_dup_exec.start()
+        return well_dup_exec.join()
+
+
+class IntegrityCheck(DemultiplexingStage):
+    previous_stages = [Bcl2FastqAndFilter]
+
+    def _run(self):
+        return executor.execute(
+            *[bash_commands.gzip_test(fq) for fq in util.find_all_fastqs(self.fastq_dir)],
+            job_name='integrity_check',
+            working_dir=self.job_dir,
+            cpus=1,
+            mem=2
+        ).join()
+
+
+class FastQC(DemultiplexingStage):
+    previous_stages = [Bcl2FastqAndFilter]
+
+    def _run(self):
+        return executor.execute(
+            *[bash_commands.fastqc(fq) for fq in util.find_all_fastqs(self.fastq_dir)],
+            job_name='fastqc',
+            working_dir=self.job_dir,
+            cpus=1,
+            mem=2
+        ).join()
+
+
+class SeqtkFQChk(DemultiplexingStage):
+    previous_stages = [Bcl2FastqAndFilter]
+
+    def _run(self):
+        return executor.execute(
+            *[bash_commands.seqtk_fqchk(fq) for fq in util.find_all_fastqs(self.fastq_dir)],
+            job_name='fqchk',
+            working_dir=self.job_dir,
+            cpus=1,
+            mem=2,
+            log_commands=False
+        ).join()
+
+
+class MD5Sum(DemultiplexingStage):
+    previous_stages = [Bcl2FastqAndFilter]
+
+    def _run(self):
+        return executor.execute(
+            *[bash_commands.md5sum(fq) for fq in util.find_all_fastqs(self.fastq_dir)],
+            job_name='md5sum',
+            working_dir=self.job_dir,
+            cpus=1,
+            mem=2,
+            log_commands=False
+        ).join()
+
+
+class QCOutput(DemultiplexingStage):
+    previous_stages = [WellDuplicates, IntegrityCheck, FastQC, SeqtkFQChk, MD5Sum]
+
+    def _run(self):
+        # Find conversion xml file and adapter file, and send the results to the rest API
+        conversion_xml = join(self.fastq_dir, 'Stats', 'ConversionStats.xml')
+        adapter_trim_file = join(self.fastq_dir, 'Stats', 'AdapterTrimming.txt')
+
+        if exists(conversion_xml) and exists(adapter_trim_file):
+            self.info('Found ConversionStats and AdaptorTrimming. Sending data.')
+            crawler = RunCrawler(
+                self.dataset.name, self.sample_sheet, adapter_trim_file=adapter_trim_file,
+                conversion_xml_file=conversion_xml, run_dir=self.fastq_dir
+            )
+            # TODO: review whether we need this
+            json_file = join(self.fastq_dir, 'demultiplexing_results.json')
+            crawler.write_json(json_file)
```

Comment:
We should drop these 3 lines now the `demultiplexing_results.json` file has never been used